CONTRIBUTING.md

```diff
@@ -9,3 +9,27 @@ https://qiskit.org/documentation/contributing_to_qiskit.html
 
 In addition to the general guidelines there are specific details for
 contributing to qiskit_sphinx_theme, these are documented below.
+
+# Folder Structure
+
+There are a few important subfolders to be aware of:
+
+## `/qiskit_sphinx_theme`
+This subfolder contains the raw html and css files that are used to set the framework for Qiskit documentation web pages. The files in this folder will set the styles when a project calls the `qiskit_sphinx_theme` in it's `conf.py`, unless the project overrides the style with their own files.
+
+E.g. if a qiskit project (with the `qiskit_sphinx_theme` installed) wants to override the `layout.html` that the `qiskit_sphinx_theme` provides it needs to have it's own `layout.html` stored in its `docs/_templates` folder.
+
+## `/docs`
+This subfolder contains a scaled down version fo the Sphinx build that builds the documentation for the Qiskit repos. It pulls styles from the `/qiskit_sphinx_theme` subfolder. You can check any changes you are making to the `qiskit_sphinx_theme` by building the documentation (see running locally section) and opening the html files generated in `/docs/_build/html`
+
+## `/docs/core_pages`
```

Comment:
I'm a bit confused about this is this just for testing the theme? If we want to do a docs build with local changes applied it would probably be easier to install the theme from source and just clone `Qiskit/qiskit` and build the full docs in CI. It'll be a bit slower but it'll give a better picture than trying to figure out how to maintain a local copy of a subset of the out of tree pages

Reply:
I think for making changes to the theme it makes sense to keep a mini sphinx build here, just so we can quickly see any UI changes we're making for things like the top bar and side bar. and then i can also test out how potential changes in a `conf.py` would effect things without doing a v long build. For UI/styling changes it helps to have as quick a build as possible.

As for the core pages we talked about moving some of the main pages (contributors guide, deprecation policy etc.) out of the metapackage and into this repo so we can deploy them straight from here. I'm not sure yet if it would be better to rewrite them all in pure html, or have this mini sphinx build generate the html from the .rst files (maybe we can discuss the COs deployment stuff more when you're back from vacation?). In the meantime I just put the files here in their `.rst` form, but perhaps it would be better to move them out of the /docs subdir until we decide how to handle them 🤔